### PR TITLE
Feature/new users

### PR DESCRIPTION
**Pull request by ADavidson02**
## What is the change?
* Functionality was added to add new users

## Is this a fix or a feature? 
* Feature

## Where should the reviewer start?
* SignUp.js line 1

## How should this be tested?
run all tests and check they pass
* From the main screen press the "sign up" button 
* Fill in all fields then press sign up
* from there you should be pushed to the dashboard and met with a message that you do not have any dream data

## Relevant Screenshots:
Screen Shot 2021-03-10 at 2 48 17 PM
Screen Shot 2021-03-10 at 2 47 41 PM
Screen Shot 2021-03-10 at 2 48 01 PM

### CODE REVIEW

File: src/Analytics/TonesOverTime.js [hunk 1]
```diff
@@ -228,31 +237,38 @@ const TonesOverTime = () => {
...
     },
   };
-
+   if (noCharts === true ) {
+     return(
+        <h4>
+          You do not have any dream data, add dreams to see your dream
+          tones
+        </h4> 
+     )
```
Comment (Shawntru) [inline src/Analytics/TonesOverTime.js:246]: I like this, will be nice if we decide to add any sort of graphic or anything here later.

**Review by Shawntru — COMMENTED** [review]
Looking like user is not logged in successfully after signup, logging out and then back in again corrects it. Might be an issue with server response code, createUser method on SignUp.js line 66 might be an interesting place to start.

File: src/Analytics/TonesOverTime.js (hunk 1 above)
Comment (ADavidson02) [reply src/Analytics/TonesOverTime.js:246]: Agree but for now this is a good place holder

File: src/modules/SignUp/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    setSignUpError(false);
+    setduplicateWarning(false);
+    API.createNewUser(values.name, values.email, values.password).then(
+      (response) => {
+        if (response.status === 409) {
+          setDisabled(true);
+          setduplicateWarning(true);
+          return;
+        } else if (response.email === values.email) {
+          loginUser(values.email, values.password);
```
Comment (ADavidson02) [inline src/modules/SignUp/SignUp.js:92]: @Shawntru here is where I added the login, this seems to fix the problem of the new user not being signed in and not adding the new dreams on first login.

File: src/modules/DreamJournal/DreamJournal.js [hunk 3]
```diff
@@ -44,7 +44,8 @@ const DreamJournal = () => {
     setLoading(true);
     setDreamsError(false);
     API.fetchUserDreams(user.token).then((response) => {
-      if (response === []) {
+      console.log(response)
```
Comment (Shawntru) [inline src/modules/DreamJournal/DreamJournal.js]: Can we remove this cl?

**Review by Shawntru — APPROVED** [review]
Fixes look great! Got a console.log in there that can be removed if you want. Nice work on this, looks awesome  :)

**Review by ADavidson02 — COMMENTED** [review]
> Fixes look great! Got a console.log in there that can be removed if you want. Nice work on this, looks awesome :)

Removed thank you for catching this

**Pull request merged**